include/sleip/dynamic_array.hpp

```diff
@@ -0,0 +1,320 @@
+#ifndef SLEIP_DYNAMIC_ARRAY_HPP_
+#define SLEIP_DYNAMIC_ARRAY_HPP_
+
+#include <boost/core/alloc_construct.hpp>
+#include <boost/throw_exception.hpp>
+#include <boost/config.hpp>
+
+#include <cstddef>
+#include <iterator>
+#include <memory>
+#include <utility>
+#include <type_traits>
+
+namespace sleip
+{
+template <class T, class Allocator>
+struct dynamic_array;
+
+template <class T, class Allocator = std::allocator<T>>
```

Comment:
Any reason to put the default argument here instead of at the declaration?

Reply:
I actually wasn't sure what was considered the "normal" thing to do here. Will move to the declaration.